docs/contributing/running-and-writing-acceptance-tests.md

```diff
@@ -1136,47 +1162,248 @@ The below are style-based items that _may_ be noted during review and are recomm
-- [ ] __Uses Builtin ARN Check Functions__: Tests should utilize available ARN check functions, e.g. `testAccMatchResourceAttrRegionalARN()`, to validate ARN attribute values in the Terraform state over `resource.TestCheckResourceAttrSet()` and `resource.TestMatchResourceAttr()`
-- [ ] __Uses testAccCheckResourceAttrAccountID()__: Tests should utilize the available AWS Account ID check function, `testAccCheckResourceAttrAccountID()` to validate account ID attribute values in the Terraform state over `resource.TestCheckResourceAttrSet()` and `resource.TestMatchResourceAttr()`
+
+Here's an example of using `aws_region` and `data.aws_region.current.name`:
+
+```hcl
+data "aws_region" "current" {}
+
+resource "aws_route53_zone" "test" {
+  vpc {
+    vpc_id     = aws_vpc.test.id
+    vpc_region = data.aws_region.current.name
+  }
+}
+```
+
+#### Hardcoded Spot Price
+
+- [ ] __Uses aws_ec2_spot_price Data Source__: Any hardcoded spot prices, e.g. `0.05`, should be replaced with the [`aws_ec2_spot_price` data source](https://www.terraform.io/docs/providers/aws/d/ec2_spot_price.html). A common pattern is declaring `data "aws_ec2_spot_price" "current" {}` and referencing it via `data.aws_ec2_spot_price.current.spot_price`.
+
+Here's an example of using `aws_ec2_spot_price` and `data.aws_ec2_spot_price.current.spot_price`:
+
+```hcl
+data "aws_ec2_spot_price" "current" {
+  instance_type = "t3.medium"
+
+  filter {
+    name   = "product-description"
+    values = ["Linux/UNIX"]
+  }
+}
+
+resource "aws_spot_fleet_request" "test" {
+  spot_price      = data.aws_ec2_spot_price.current.spot_price
+  target_capacity = 2
+}
+```
+
+#### Hardcoded State Hash
+
+- [ ] __Uses tfawsresource.TestCheckTypeSetElemNestedAttrs()__: Any hardcoded state hashes used in test steps, e.g., `resource.TestCheckResourceAttr(resourceName, "port_range.4120185872.from_port", "80")`, should be replaced with `tfawsresource.TestCheckTypeSetElemNestedAttrs()`.
+
+Here's an example of using `TestCheckTypeSetElemNestedAttrs()`:
+
+```go
+func TestAccAwsGlobalAcceleratorListener_basic(t *testing.T) {
+	resourceName := "aws_globalaccelerator_listener.example"
+	rInt := acctest.RandInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckGlobalAccelerator(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGlobalAcceleratorListenerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlobalAcceleratorListener_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGlobalAcceleratorListenerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "client_affinity", "NONE"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "port_range.*", map[string]string{
+						"from_port": "80",
+						"to_port":   "81",
+					}),
+				),
+			},
+		},
+	})
+}
+```
```

Comment:
It might be best to leave this section out of our documentation here since these helpers have now been migrated into the Terraform Plugin SDK and ensure the "Extending Terraform" documentation includes how to use them.

```suggestion
```